activitysim/abm/models/trip_purpose.py

```diff
@@ -61,36 +63,40 @@ def choose_intermediate_trip_purpose(trips, probs_spec, estimator, trace_hh_id,
     probs_spec.loc[:, purpose_cols] = probs_spec.loc[:, purpose_cols].div(sum_probs, axis=0)
 
     # left join trips to probs (there may be multiple rows per trip for multiple depart ranges)
-    choosers = pd.merge(trips.reset_index(), probs_spec, on=PROBS_JOIN_COLUMNS,
+    choosers = pd.merge(trips.reset_index(), probs_spec, on=probs_join_cols,
                         how='left').set_index('trip_id')
     chunk.log_df(trace_label, 'choosers', choosers)
 
-    # select the matching depart range (this should result on in exactly one chooser row per trip)
-    chooser_probs = \
-        (choosers.start >= choosers['depart_range_start']) & (choosers.start <= choosers['depart_range_end'])
-
-    # if we failed to match a row in probs_spec
-    if chooser_probs.sum() < num_trips:
-        # this can happen if the spec doesn't have probs for the trips matching a trip's probs_join_cols
-        missing_trip_ids = trips.index[~trips.index.isin(choosers.index[chooser_probs])].values
-        unmatched_choosers = choosers[choosers.index.isin(missing_trip_ids)]
-        unmatched_choosers = unmatched_choosers[['person_id', 'start'] + non_purpose_cols]
-
-        # join to persons for better diagnostics
-        persons = inject.get_table('persons').to_frame()
-        persons_cols = ['age', 'is_worker', 'is_student', 'is_gradeschool', 'is_highschool', 'is_university']
-        unmatched_choosers = pd.merge(unmatched_choosers, persons[persons_cols],
-                                      left_on='person_id', right_index=True, how='left')
-
-        file_name = '%s.UNMATCHED_PROBS' % trace_label
-        logger.error("%s %s of %s intermediate trips could not be matched to probs based on join columns  %s" %
-                     (trace_label, len(unmatched_choosers), len(choosers), PROBS_JOIN_COLUMNS))
-        logger.info("Writing %s unmatched choosers to %s" % (len(unmatched_choosers), file_name,))
-        tracing.write_csv(unmatched_choosers, file_name=file_name, transpose=False)
-        raise RuntimeError("Some trips could not be matched to probs based on join columns %s." % PROBS_JOIN_COLUMNS)
-
-    # select the matching depart range (this should result on in exactly one chooser row per trip)
-    choosers = choosers[chooser_probs]
+    if use_depart_time:
```

Comment:
What is the intended purpose of this filter?

Reply:
This makes the trip purpose model more flexible by allowing the user to specify whether or not the trip purpose probability lookups should be filtered by matching departure time ranges or not. Before, it was assumed that they should, and an error was thrown if the probability lookup table didn't have departure range columns.